### PR TITLE
I created Database interface, InMemoryDatabase class, InvoiceService …

### PR DESCRIPTION
**Pull request by pawkli95**
…class, Vat enum, Invoice class, InvoiceEntry class, Company class and according tests

### CODE REVIEW

File: app/src/main/java/pl/futurecollars/invoicing/model/Vat.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package pl.futurecollars.invoicing.model;
+
+import java.math.BigDecimal;
+
+public enum Vat {
+    V_23(new BigDecimal("0.23")),
```
Comment (michalflis) [inline app/src/main/java/pl/futurecollars/invoicing/model/Vat.java:6]: You need to put empty line here. 
Error: eckstyle] [ERROR] /home/runner/work/invoicing-system-Pawel-Klimek/invoicing-system-Pawel-Klimek/app/src/main/java/pl/futurecollars/invoicing/model/Vat.java:5: Leave empty row after class/interface/enum definition! [RegexpMultiline]

**Review by michalflis — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/futurecollars/invoicing/service/InvoiceService.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    public Optional<Invoice> getById(UUID id) {
+        return database.getById(id);
+    }
+
+    public List<Invoice> getAll() {
+        return database.getAll();
+    }
+
+    public List<Invoice> filter(Predicate<Invoice> predicate) {
```
Comment (michalflis) [inline app/src/main/java/pl/futurecollars/invoicing/service/InvoiceService.java:32]: You can simplify this method:


```
return database.getAll()
                .stream()
                .filter(predicate)
                .collect(Collectors.toList());
```

File: app/src/main/java/pl/futurecollars/invoicing/model/Vat.java (hunk 1 above)
Comment (michalflis) [reply app/src/main/java/pl/futurecollars/invoicing/model/Vat.java:6]: And you have a lot of other checkstyle violations:
  Checkstyle files with violations: 8
  Checkstyle violations by severity: [warning:44]

File: app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package pl.futurecollars.invoicing.db.memory;
+
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import java.util.*;
+
+public class InMemoryDatabase implements Database {
+
+
+    public final Map<UUID, Invoice> database = new HashMap<>();
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java:10]: This field should be private - we don't want to have access to this Map outside of our interface

File: app/src/main/java/pl/futurecollars/invoicing/model/Vat.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    V_8(new BigDecimal("0.08")),
+    V_5(new BigDecimal("0.05")),
+    V_0(BigDecimal.ZERO);
+
+    public final BigDecimal rate;
+
+
+    private Vat(BigDecimal rate) {
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/model/Vat.java:14]: BigDecimal is overkill for this use case. Float should be enough.

File: app/src/main/java/pl/futurecollars/invoicing/model/Invoice.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Data
+public class Invoice {
+
+    private UUID id = UUID.randomUUID();
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/model/Invoice.java:14]: It is better to initialize it in constructor. You can override empty constructor and set value to this variable.

File: app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            }
+            database.put(invoice.getId(), invoice);
+            return invoice;
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<Invoice> getById(UUID id) {
+        return Optional.ofNullable(database.get(id));
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java:27]: It is good practice to use `Optional` when we have a method called such as `find...`. If we have method call `get...` we shouldn't expect `null` out of it

File: app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public Invoice update(Invoice updatedInvoice) {
+        Optional<Invoice> optionalInvoice = Optional.ofNullable(updatedInvoice);
+        if(optionalInvoice.isPresent()) {
+            database.put(updatedInvoice.getId(), updatedInvoice);
+            return updatedInvoice;
+        }
+            return null;
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/db/memory/InMemoryDatabase.java:42]: weird code format

**Review by radoslawdabrowski — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/futurecollars/invoicing/model/Vat.java (hunk 4 above)
Comment (michalflis) [reply app/src/main/java/pl/futurecollars/invoicing/model/Vat.java:14]: I am applying this suggestion in my Enum also.

**Pull request closed without merging**